packages/config/src/ProfileInfo.ts

```diff
@@ -1150,7 +1151,9 @@ export class ProfileInfo {
             }
         } else {
             // Load profile schemas from meta files in profile root dir
-            for (const { type } of this.mOldSchoolProfileCache) {
+            const profTypes = fs.existsSync(this.mOldSchoolProfileRootDir) ?
+                ProfileIO.getAllProfileDirectories(this.mOldSchoolProfileRootDir) : [];
+            for (const type of profTypes) {
```

Comment:
I see that this is the same technique that we use in `readProfilesFromDisk`.

I'm curious if it is necessary to do this in order to avoid a few extra `fs.existsSync` calls?
OR
if we can just add a `ensureReadFromDisk` call at the top of the `loadAllSchemas` and that way we can continue to assume that `mOldSchoolProfileCache` will have everything in it (OR will be an empty array)

🤔 It feels like calling `getAllProfileDirectories` here may not save us much.

I'd like to chat a bit about this one though 😋